api/internal/localizer/localizer.go

```diff
@@ -289,6 +333,13 @@ func (lc *localizer) localizeResource(path string) (string, error) {
 	return locPath, nil
 }
 
+func (lc *localizer) localizeFileIfSet(entry string) (string, error) {
+	if entry == "" {
+		return "", nil
+	}
+	return lc.localizeFile(entry)
+}
```

Comment:
super nit: can we merge this function into `localizeFile`.

Reply:
Agree that folding it in is cleaner. If you want to ignore errors for empty entry in some cases but not others, you could add an additional parameter to `localizeFile`, like `localizeFile(path string, errIfEmpty bool)`.

Reply:
Done. I added the check for the empty string to `localizeFile`. 

Thanks for the suggestion, @natasha41575. I considered this, but want to keep the method signature of `localizeFile` as `func(string) (string, error)` for ease of use `localizeBuiltinPlugins`' `locPathFn func(string) (string, error)` in `builtinplugins.go` file. I also reasoned that localize doesn't have to throw errors on empty entries since `kustomize build` will take care of this and we don't have to re-implement `build` functionality. 

However, to keep treatment of empty fields consistent, I added the check for empty paths to `localizeRoot`: https://github.com/kubernetes-sigs/kustomize/pull/4978/files#diff-25faa786bf35469d391d5eef99ed1323623023442b177c1ed3e07f3e6709dac6R385-R387
I also added the following test to demonstrate localize's consistent treatment of empty paths: https://github.com/kubernetes-sigs/kustomize/pull/4978/files#diff-c21f7b3d0afda4a64163411e90e72d2fed6c3df14c7b6f8ed07237bb5e1fc91cR1317